Source/WebCore/platform/graphics/cocoa/VideoMediaSampleRenderer.mm

```diff
@@ -344,22 +339,26 @@ - (void)resetUpcomingSampleBufferPresentationTimeExpectations;
     assertIsMainThread();
     [renderer() flush];
 
-    if (m_decompressionSession)
-        m_decompressionSession->flush();
-
-    m_workQueue->dispatch([weakThis = ThreadSafeWeakPtr { *this }] () mutable {
-        RefPtr protectedThis = weakThis.get();
-        if (!protectedThis)
-            return;
-
-        protectedThis->flushCompressedSampleQueue();
-        protectedThis->flushDecodedSampleQueue();
-
-        callOnMainThread([weakThis = WTFMove(weakThis)] {
+    auto checkReadyForMoreMediaData = [weakThis = ThreadSafeWeakPtr { *this }]() mutable {
+        ensureOnMainThread([weakThis = WTFMove(weakThis)] {
             if (RefPtr protectedThis = weakThis.get())
                 protectedThis->maybeBecomeReadyForMoreMediaData();
         });
+    };
+
+    if (!m_decompressionSession) {
+        checkReadyForMoreMediaData();
```

Comment:
Previously we would call maybeBecomeReadyForMoreMediaData asynchronously.
Now we may call it synchronously with flush, which could lead to more reentrancy.
Does it matter?

Reply:
Actually, the call to `maybeBecomeReadyForMoreMediaData` is an addition that was added with PR #35218, it's not needed and should only be used with a decompression session.